app/views/games/_chessboard.html.erb

```diff
@@ -34,6 +37,19 @@ $(function() {
           alert("Invalid Move");
         }
       });
+       ( function () {
+       $.get( '/pieces/' + update_piece.id,
+         dataType('json')).success(function(data){
+          var h3Element = $('h3');
+        console.log("goodbye");
+// if (data  = black)  {
```

Comment:
So, this looks like a good strategy, you'll just need to pass the current player data in the controller response, so something like:

```ruby
render json: { success: true, current_player: @piece.game.current_player == @piece.game.black_player ? 'black' : 'white' }

# you could extract that condition into a new method on game if you like, something like currently_black_turn? or whatever
# then you could do:
render json: { success: true, current_player: @piece.game.currently_black_turn? ? 'black' : 'white' }

# you'd also need to change the failure responses to:
render json: { success: false }
```

Then here you can do:

```javascript
if (data.success) {
  if (data.current_player === 'black') {
    h3Element.html('Black Player Turn');
  } else {
    h3Element.html('White Player Turn');
  }
}
```

Keep in mind that you have to handle both paths, black and white, because even though it's already set up for white above, that might get changed and then need to be changed back.